src/snowflake/cli/_plugins/nativeapp/entities/application.py

```diff
@@ -569,23 +564,23 @@ def execute_post_deploy_hooks(self):
         workspace_ctx = self._workspace_ctx
         console = workspace_ctx.console
         project_root = workspace_ctx.project_root
+        app_role = (model.meta and model.meta.role) or workspace_ctx.default_role
+        app_warehouse = (
+            model.meta and model.meta.warehouse
+        ) or workspace_ctx.default_warehouse
```

Comment:
you should rebase on top of Frank's latest work (expect conflicts though, but it will make your job easier overall)